python/cuda_parallel/cuda/parallel/experimental/iterators/_strided.py

```diff
@@ -0,0 +1,240 @@
+import ctypes
+from functools import lru_cache
+from typing import Tuple
+
+import numba
+import numpy as np
+from numba import types
+from numba.core import cgutils
+from numba.core.extending import (
+    models,
+    register_model,
+)
+from numba.core.typing.templates import AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.cuda.cudaimpl import registry as cuda_lower_registry
+
+import cuda.parallel.experimental.iterators._iterators as _iterators
+
+
+@lru_cache
+def make_iterator_struct_class(ndim):
+    shape_ctype = ctypes.c_int64 * ndim
+    strides_ctype = ctypes.c_int64 * ndim
+
+    class StridedArrayView(ctypes.Structure):
+        _fields_ = [
+            ("linear_id", ctypes.c_int64),
+            ("ptr", ctypes.c_void_p),
+            ("shape", shape_ctype),
+            ("strides", strides_ctype),
+            ("ndim", ctypes.c_int32),
+        ]
+
+    return StridedArrayView
+
+
+def iterator_struct_ctype(ptr: int, ndim: int, shape: Tuple[int], strides: Tuple[int]):
+    StridedArrayView_cls = make_iterator_struct_class(ndim)
+
+    c_shape = (ctypes.c_int64 * ndim)(*shape)
+    c_strides = (ctypes.c_int64 * ndim)(*strides)
+    return StridedArrayView_cls(0, ptr, c_shape, c_strides, ndim)
+
+
+@lru_cache
+def strided_view_iterator_numba_type(value_type: types.Type, ndim: int):
```

Comment:
This seems like a useful machinery that we could reuse for numba-cuda to recognize `cuda.core.StridedMemoryView`. Let's discuss this in an ongoing internal chat.

Reply:
xref: https://github.com/NVIDIA/numba-cuda/issues/153